.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,28 @@
+// For format details, see https://aka.ms/devcontainer.json. 
+// For config options, see https://github.com/devcontainers/templates/tree/main/src/ubuntu
+{
+  "name": "Ubuntu",
+  "image": "mcr.microsoft.com/devcontainers/base:jammy",
+  "customizations": {
```

Comment:
How is this going to work for arm builds? Will we use cross compilation and a toolchain or would we need to run this on an ARM host?

Also, what if I need a custom devcontainer for my project that requires additional volumes or other features?

Reply:
Good question, I haven't taken a shot at ARM builds, but I have a much better sense than before of how to express cross-compilation cleanly in CMake. Running on an ARM builder is also always an option. 

There is also the vcpkg support to consider. From a quick glance it looks like the public vcpkg repository does not include arm64-linux triplets in its CI (it does include arm64-windows). So to make vcpkg play well with arm64-linux we might need to use its "[overlay triplets](https://learn.microsoft.com/en-us/vcpkg/concepts/triplets)" feature?

We could definitely make derisking ARM builds a condition of pursuing this approach.

> Also, what if I need a custom devcontainer for my project that requires additional volumes or other features?

I think VsCode supports [specifying / switching between multiple devcontainers](https://github.com/microsoft/vscode-remote-release/issues/1165#issuecomment-1491325370). 

As you know you can specify the devcontainer as a Dockerfile; this is just using a higher-level API to show one example of a basic devcontainer.